client.go

```diff
@@ -118,7 +125,13 @@ func (c *Castle) sendCall(ctx context.Context, r *castleAPIRequest, url string)
 	}()
 
 	b := new(bytes.Buffer)
-	err = json.NewEncoder(b).Encode(r)
+
+	switch request := r.(type) {
+	case *castleRiskAPIRequest:
+		err = json.NewEncoder(b).Encode(request)
+	case *castleFilterAPIRequest:
+		err = json.NewEncoder(b).Encode(request)
```

Comment:
```suggestion
	case *castleRiskAPIRequest:
	case *castleFilterAPIRequest:
		err = json.NewEncoder(b).Encode(request)
```

Reply:
Can do just that?

Reply:
or commas https://go.dev/wiki/Switch#multiple-cases

Reply:
have chosen commas for now, had a look at generics but I honestly haven't quite grasped them yet so I may have missed something.

Reply:
https://github.com/utilitywarehouse/castle-go/pull/27/files

^ for illustration purposes